File: jsweet-client/src/main/java/json_schema/states/BotState.java
package json_schema.states;

import static def.jquery.Globals.$;

import json_schema.types.Point;

public class BotState extends def.js.Object {

	public Double getEnergyLevel() {
		return (Double) $get("energy");
	}

	public Point getPosition() {
		return (Point) $.extend(false, new Point(), $get("position"));
	}

	public Double getDirection() {
		return (Double) $get("direction");
	}

	public Double getGunDirection() {
		return (Double) $get("gunDirection");
	}

	public Double getRadarDirection() {
		return (Double) $get("radarDirection");
	}

	public Double getRadarSpreadAngle() {
		return (Double) $get("radarSpreadAngle");
	}

	public Double getSpeed() {
		return (Double) $get("speed");
	}
}

File: robocode2-gui/src/main/kotlin/net/robocode2/gui/model/event/BotState.kt
package net.robocode2.gui.model.event

import net.robocode2.gui.model.types.Point

data class BotState(
        val id: Int,
        val energy: Double,
        val position: Point,
        val direction: Double,
        val gunDirection: Double,
        val radarDirection: Double,
        val radarSweep: Double,
        val speed: Double
)